src/pages/BasePersonalisationPage.js

```diff
@@ -0,0 +1,51 @@
+/* @flow */
+
+"use strict";
+
+import React from 'react';
+import Router from "react-router";
+import mui from "material-ui";
+import reactMixin from "react-mixin";
+import _ from "underscore";
+
+import BaseCategoriesPage from './BaseCategoriesPage';
+import components from '../components';
+import icons from '../icons';
+
+/*::`*/@reactMixin.decorate(Router.Navigation)/*::`;*/
+/*::`*/@reactMixin.decorate(Router.State)/*::`;*/
+class BasePersonalisationPage extends BaseCategoriesPage {
+    constructor(props: Object) {
+        super(props);
+        this.state = {
+            subpage: 0,
+        };
+    }
+
+    // flow:disable
+    static childContextTypes = {
+        controller: React.PropTypes.instanceOf(BasePersonalisationPage),
+    };
+
+    getChildContext(): Object {
+        return {
+            controller: this,
+        };
+    }
+
+    previousStep(): void {
```

Comment:
We want a state graph that looks like this:

```
[Home] -> [Intro] -> [Questions] -> [Location] -> [Results]
       ^--------------------------------------------------------/
```

So we're using on click handlers to drive the state machine.